server/webapp/WEB-INF/rails.new/app/helpers/application_helper.rb

```diff
@@ -103,6 +103,10 @@ def random_dom_id(prefix = "")
     prefix + java.util.UUID.randomUUID().to_s
   end
 
+  def sanitize_for_dom_id(value)
+    value.gsub('.', '_dot_').tr('^a-zA-Z0-9_-', '_')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.